MozillaFoundation/foundation.mozilla.org
# Add `bgcolor` and `textcolor` fields to `DonateBanner` model

# Description

This PR updates the `DonateBanner` model with the addition of two new fields `background_color` and `text_color`, allowing the user to customize the donate banner present on all site pages. It also renames the previously incorrectly named `background_image` to `foreground_image` and adds a `background_image` field to store the newly implemented background image feature for the banner.

# To test

1. Open the review app [Wagtail auth page](https://foundation-s-tp1-1234-i-bhpnku.herokuapp.com/cms/), authenticate with `admin2:admin2` credentials.
2. Navigate to the `Settings > Donate banner` [page](https://foundation-s-tp1-1234-i-bhpnku.herokuapp.com/cms/settings/donate_banner/sitedonatebanner/1/), click on the `Choose` in order to pick a banner to show.
<img width="521" alt="image" src="https://github.com/user-attachments/assets/c96e684d-3916-4bbf-94a8-98ef7eb829e7">

3. Select a banner from the list by clicking on its name. Save by clicking on the `Save` button on the bottom of the page.
<img width="536" alt="image" src="https://github.com/user-attachments/assets/f95b631c-b60a-4223-ab1f-59ff2153b60f">
<img width="392" alt="image" src="https://github.com/user-attachments/assets/f7b156f9-fab9-47fd-8c18-b914f49d5486">

4. Edit the selected banner by clicking on the three dots next to its name and then selecting the `Edit` menu:
<img width="640" alt="image" src="https://github.com/user-attachments/assets/b20a1d18-0db1-46f6-92b0-3f9875e78247">

5. Note the new model fields `Background image`, `Background color` and `Text color`. Customize the background color of the banner. Save and browse the [preview app homepage](https://foundation-s-tp1-1234-i-hlbzy3.herokuapp.com/en/).
![image](https://github.com/user-attachments/assets/e910508a-72f3-40b4-bb91-5fec611fb8ab)

6. Confirm that the background color has changed in the [homepage](https://foundation-s-tp1-1234-i-bhpnku.herokuapp.com/en/). 
<img width="1350" alt="image" src="https://github.com/user-attachments/assets/912488ac-1f39-4563-b7aa-d747808c8d46">

7. Edit again the donate banner and upload a background image. Save and confirm changes reflect on the [homepage](https://foundation-s-tp1-1234-i-bhpnku.herokuapp.com/en/).
<img width="1337" alt="image" src="https://github.com/user-attachments/assets/790d5bb8-f2e6-429b-86b4-dc237650ec96">


Link to sample test page: https://foundation-s-tp1-1234-i-bhpnku.herokuapp.com/en/
Related PRs/issues: #12881

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-1360)
